pkg/morph/deploy/notary.go

```diff
@@ -0,0 +1,880 @@
+package deploy
+
+import (
+	"bytes"
+	"context"
+	"crypto/sha256"
+	"encoding/base64"
+	"encoding/binary"
+	"errors"
+	"fmt"
+
+	"github.com/nspcc-dev/neo-go/pkg/core/native/noderoles"
+	"github.com/nspcc-dev/neo-go/pkg/core/transaction"
+	"github.com/nspcc-dev/neo-go/pkg/crypto/keys"
+	"github.com/nspcc-dev/neo-go/pkg/rpcclient/actor"
+	"github.com/nspcc-dev/neo-go/pkg/rpcclient/invoker"
+	"github.com/nspcc-dev/neo-go/pkg/rpcclient/nns"
+	"github.com/nspcc-dev/neo-go/pkg/rpcclient/rolemgmt"
+	"github.com/nspcc-dev/neo-go/pkg/smartcontract"
+	"github.com/nspcc-dev/neo-go/pkg/util"
+	"github.com/nspcc-dev/neo-go/pkg/vm/opcode"
+	"github.com/nspcc-dev/neo-go/pkg/wallet"
+	randutil "github.com/nspcc-dev/neofs-node/pkg/util/rand"
+	"go.uber.org/zap"
+)
+
+// enableNotaryPrm groups parameters of Notary service initialization parameters
+// for the committee.
+type enableNotaryPrm struct {
+	logger *zap.Logger
+
+	blockchain Blockchain
+
+	nnsOnChainAddress util.Uint160
+	systemEmail       string
+
+	committee              keys.PublicKeys
+	localAcc               *wallet.Account
+	localAccCommitteeIndex int
+}
+
+// enableNotary makes Notary service ready-to-go for the committee members.
+func enableNotary(ctx context.Context, prm enableNotaryPrm) error {
+	monitor, err := newBlockchainMonitor(prm.logger, prm.blockchain)
+	if err != nil {
+		return fmt.Errorf("init blockchain monitor: %w", err)
+	}
+	defer monitor.stop()
+
+	var tick func()
```

Comment:
because it's defined conditionally (may be not needed)

Reply:
IMO it will be easier to read if we have definition logic separately (the `for` below is already complex)

Reply:
definition or initialization? u suggest to compose multi-tick context before loop?

Reply:
> definition or initialization?

i see no difference in that context. initialization, i think

> u suggest to compose multi-tick context before loop?

i saw that `var` and the `for` below looked more complex than they could be: you declared a var and defined it in the place where it should be used (also defined it more than one time in some cases cause it is a `for` loop). its definition (initialization) does not need to be flexible (it only returns an error if `actor` can't be created, quite a static thing). so my question is: can it be initialized once (or two vars created (or an `actor` and just a static func that is a closure over just created `actor`)) instead of `var v func()` and `v = someStaticThingThatAlwayWillBeTheSame()` in every iteration?

Reply:
logic is simple: `tick` can be not needed at all, so it's initialized once under certain condition in for loop. Do u suggest to insta initialize it outside loop?

Reply:
> Do u suggest to insta initialize it outside loop?

yes, that is the only thing i suggest. im not sure it is possible cause func is quiete complex and i do not keep it in mind completely.
OR i think it can be called in every `if` or `else`. not sure we need them (`if` and `else`) to instantiate smth that will be called anyway in the end (outside of every `if` and `else`) of every iteration

Reply:
done